From 9bd20b7304aae61de5d50ac359cf27132bafd4c1 Mon Sep 17 00:00:00 2001
From: Werner Lemberg <wl@gnu.org>
Date: Sat, 22 Nov 2014 06:24:45 +0100
Subject: [PATCH] [sfnt] Fix Savannah bug #43656.

* src/sfnt/ttcmap.c (tt_cmap4_validate): Fix order of validity
tests.

Upstream-Status: Backport

Signed-off-by: Yue Tao <yue.tao@windriver.com>
---
 ChangeLog         |    7 +++++++
 src/sfnt/ttcmap.c |    6 +++---
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index ee27f70..150f65a 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -35,6 +35,13 @@
 
 2014-11-22  Werner Lemberg  <wl@gnu.org>
 
+	[sfnt] Fix Savannah bug #43656.
+
+	* src/sfnt/ttcmap.c (tt_cmap4_validate): Fix order of validity
+	tests.
+
+2014-11-22  Werner Lemberg  <wl@gnu.org>
+
 	[cff] Fix Savannah bug #43658.
 
 	* src/cff/cf2ft.c (cf2_builder_lineTo, cf2_builder_cubeTo): Handle
diff --git a/src/sfnt/ttcmap.c b/src/sfnt/ttcmap.c
index 712bd4f..fb863c3 100644
--- a/src/sfnt/ttcmap.c
+++ b/src/sfnt/ttcmap.c
@@ -845,9 +845,6 @@
     p      = table + 2;           /* skip format */
     length = TT_NEXT_USHORT( p );
 
-    if ( length < 16 )
-      FT_INVALID_TOO_SHORT;
-
     /* in certain fonts, the `length' field is invalid and goes */
     /* out of bound.  We try to correct this here...            */
     if ( table + length > valid->limit )
@@ -858,6 +855,9 @@
       length = (FT_UInt)( valid->limit - table );
     }
 
+    if ( length < 16 )
+      FT_INVALID_TOO_SHORT;
+
     p        = table + 6;
     num_segs = TT_NEXT_USHORT( p );   /* read segCountX2 */
 
-- 
1.7.9.5

